components/tidb_query_common/src/storage/scanner.rs

```diff
@@ -1,9 +1,19 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
+use std::time::Duration;
+
+use tikv_util::time::Instant;
+use yatp::task::future::reschedule;
+
 use super::{range::*, ranges_iter::*, OwnedKvPair, Storage};
 use crate::error::StorageError;
 
 const KEY_BUFFER_CAPACITY: usize = 64;
+/// Batch executors are run in coroutines. `MAX_TIME_SLICE` is the maximum time
+/// a coroutine can run without being yielded.
+const MAX_TIME_SLICE: Duration = Duration::from_millis(1);
```

Comment:
MAX_TIME_SLICE better to be configurable.

Reply:
Maybe we can change `MAX_TIME_SLICE` to be configurable in a separate pr if need. 😁